include/battle.h

```diff
@@ -1243,4 +1323,170 @@ static inline bool32 IsBattlerInvalidForSpreadMove(u32 battlerAtk, u32 battlerDe
         || (battlerDef == BATTLE_PARTNER(battlerAtk) && (moveTarget == MOVE_TARGET_BOTH));
 }
 
+static inline u32 MaxRaiseOrLowerStatAmount(u32 battler, u32 stat, bool32 lowering)
+{
+    return lowering ? (gBattleMons[battler].statStages[stat] - MIN_STAT_STAGE) : (MAX_STAT_STAGE - gBattleMons[battler].statStages[stat]);
+}
+
+static inline union StatChanger PrepareStatChangerAny(union StatFlags stats, s32 stage, bool32 doNotSetStatId)
+{
+    union StatChanger statChanger = (union StatChanger) {
+        .isNegative = (stage < 0),
+        .statId = (doNotSetStatId || (popcount(stats.allStats) > 1)) ? 0 : log2Int(stats.allStats),
+        .attack = stats.attack, // All 1 or 0 depending on if the stat is selected in 'stats'
+        .defense = stats.defense,
+        .speed = stats.speed,
+        .spAttack = stats.spAttack,
+        .spDefense = stats.spDefense,
+        .accuracy = stats.accuracy,
+        .evasion = stats.evasion,
+    };
+
+    // Apply stat stage and return
+    statChanger.allStats *= min(MAX_STAT_STAGE, abs(stage));
+
+    return statChanger;
+}
+
+static inline union StatChanger CalcStatChangerValue(u32 statId, s32 stage)
+{
+    return PrepareStatChangerAny(1 << statId, stage, FALSE);
+}
```

Comment:
Oh, since this is the only caller of `PrepareStatChangerAny` and it always passes a single stat then `stats` is known to contain exactly one bit and so instead of passing `1 << statId` and having to `popcount`/`log2Int` to extract the `statId` we could simply assign `.statId = statId`.

Reply:
Yup, just as you said - made it so that statId is an argument